src/qt/trafficgraphwidget.cpp

```diff
@@ -63,6 +66,54 @@ void TrafficGraphWidget::paintPath(QPainterPath &path, QQueue<float> &samples)
     }
 }
 
+void TrafficGraphWidget::UpdateToolTip(QMouseEvent *event, bool fShiftLeft/*=false*/)
+{
+    static int x = -1;
+    static int global_x = 0;
+    static int y = 0;
+    static int global_y = 0;
+    static int last_x = -1;
+    static int last_y = -1;
+    if (event) {
+        x = event->x();
+        global_x = event->globalX();
+        y = event->y();
+        global_y = event->globalY();
+    }
+    if (x == -1) return;
+
+    int w = width() - XMARGIN * 2;
+    int sampleCount = vTimeStamp.size();
+    int real_i = (w + XMARGIN - x) * DESIRED_SAMPLES / w;
+    static int i = real_i;
+    int new_x = x;
+    if (x == last_x && y == last_y) { // Follow ToolTip value if mouse has not moved
+        if (fShiftLeft && i>=0 && i < sampleCount) i++;
```

Comment:
FYI - the reason this code is here is to add the possibility to draw on the graph the point that the tooltip is referring to. But this could be added in a later PR.